include/picongpu/plugins/shadowgraphy/ShadowgraphyHelper.hpp

```diff
@@ -0,0 +1,777 @@
+                 */
+                auto getFourierBuf(int index)
+                {
+                    auto retBufferF = std::make_shared<HostBuffer<std::complex<float_64>, DIM3>>(
+                        DataSpace<DIM3>(getSizeX(), getSizeY(), getNumOmegas() / 2));
+                    auto dataBox = retBufferF->getDataBox();
+
+                    // The fields are split into 2 parts in the output, because the omega-domain
+                    // is not necessarily continuous due to band-pass filters
+                    vec3c* retField;
+                    if(index <= 1)
+                        retField = &ExOmega;
+                    else if(index <= 3)
+                        retField = &EyOmega;
+                    else if(index <= 5)
+                        retField = &BxOmega;
+                    else if(index <= 7)
+                        retField = &ByOmega;
+
+                    for(int i = 0; i < getSizeX(); ++i)
+                    {
+                        for(int j = 0; j < getSizeY(); ++j)
+                        {
+                            for(int o = 0; o < getNumOmegas() / 2; ++o)
+                            {
+                                int const oSigned = ((index % 2) == 0) ? o : o + getNumOmegas() / 2;
+                                dataBox({i, j, o}) = static_cast<std::complex<float_64>>((*retField)[i][j][oSigned]);
+                            }
+                        }
+                    }
+
+                    return retBufferF;
+                }
+
+
+                /** Calculate angular frequency in SI units
+                 *
+                 * @param i frequency index for trimmed plugin array
+                 *
+                 * @return angular frequency in SI units
+                 */
+                float_X omega(int i) const
+                {
+                    float_X const actualStep = dt;
+                    return 2.0_X * float_X(PI) * (float_X(i) - float_X(pluginNumT) / 2.0_X) / float_X(pluginNumT)
+                        / actualStep;
+                }
+
+
+                //! Return size of trimmed arrays in omega dimension
+                int getNumOmegas() const
+                {
+                    PMACC_VERIFY_MSG(
+                        getOmegaMaxIndex() > getOmegaMinIndex(),
+                        "Shadowgraphy: omega max <= omega min is not allowed!");
+                    return 2 * (getOmegaMaxIndex() - getOmegaMinIndex());
+                }
+
+                /** Return omega index for a non trimmed frequency array.
+                 * Used for properly treating raw Fourier space data of this plugin.
+                 *
+                 * @param i frequency index for trimmed plugin array
+                 *
+                 * @return index for non-trimmed array in frequency domain
+                 */
+                int getOmegaIndex(int i) const
+                {
+                    if(i < numOmegas / 2)
```

Comment:
suggestion: define 
```c++
const N_omega_half = numOmegas / 2;
```
and use it later. It occurs quite often.

Reply:
done